src/map.cpp

```diff
@@ -2726,22 +2726,59 @@ void map::drop_items( const tripoint &p )
     // rather than disappearing if it would be overloaded
 
     tripoint below( p );
+    int height_fallen = 0;
     while( !has_floor( below ) ) {
         below.z--;
+        height_fallen++;
     }
 
     if( below == p ) {
         return;
     }
 
-    for( const item &i : items ) {
-        // TODO: Bash the item up before adding it
-        // TODO: Bash the creature, terrain, furniture and vehicles on the tile
+    float damage = 0.0f;
+    for( item &i : items ) {
+        units::mass wt_dropped = i.weight();
+        damage += 10 * to_kilogram( wt_dropped ) * height_fallen;
+
         add_item_or_charges( below, i );
     }
 
-    // Just to make a sound for now
-    bash( below, 1 );
+    // Bash creature standing below
+    Creature *creature_below = get_creature_tracker().creature_at( below );
+    if( creature_below ) {
+        float dodge_mod = creature_below->dodge_roll();
+        dodge_mod = dodge_mod == 0 ? 1 : dodge_mod / 2;
+
+        int creature_hit_chance = rng( 0, 100 );
+        creature_hit_chance *= dodge_mod;
```

Comment:
Multiplying by dodge is a bit odd here, especially since there's no way to counter it.  I think throwing - dodge to set targeting precision, then basing hit chance on that will be better.

Reply:
Added character throwing as a chance modifier.